crates/red_knot_python_semantic/src/types.rs

```diff
@@ -4161,8 +4174,9 @@ impl<'db> FunctionType<'db> {
     fn internal_signature(self, db: &'db dyn Db) -> Signature<'db> {
         let scope = self.body_scope(db);
         let function_stmt_node = scope.node(db).expect_function();
-        let definition = semantic_index(db, scope.file(db)).definition(function_stmt_node);
-        Signature::from_function(db, definition, function_stmt_node)
+        let definitions = semantic_index(db, scope.file(db)).definitions(function_stmt_node);
```

Comment:
I think it would be worth a helper method on the semantic index for getting a single definition, when that's what we expect, rather than repeating the debug assertion and indexing at every call site.